pallets/xyk/src/lib.rs

```diff
@@ -2074,28 +2186,17 @@ impl<T: Config> XykFunctionsTrait<T::AccountId> for Pallet<T> {
 		let liquidity_asset_id = Pallet::<T>::get_liquidity_asset(first_asset_id, second_asset_id)?;
 
 		// Ensure user has enought liquidity tokens to burn
+		let liquidity_token_free_balance =
+			<T as Config>::Currency::free_balance(liquidity_asset_id.into(), &sender).into();
+		let liquidity_token_activated_balance =
+			LiquidityMiningActiveUser::<T>::get((&sender, &liquidity_asset_id));
+		let total_liquidity_tokens_user_owned =
+			liquidity_token_free_balance + liquidity_token_activated_balance;
```

Comment:
While this does seem safe, I would strongly recommend you to use checked math everywhere as this is a complicated pallet. and even with staking which was forked from moonbeam we had issues with unsafe math.
You needn't worry much about ensuring that all checks will pass before making changes to storage, if you use the transactional annotation.

Reply:
But make sure to keep in mind the exact flow when relying on it, as there may be some shared functions that show up in both transactional and non-transactional callers.